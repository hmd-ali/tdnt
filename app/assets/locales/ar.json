{
  "test": "test",
  "form": {
    "errors": {
      "email": {
        "required": "البريد الإلكتروني مطلوب"
      }
    }
  }
}
