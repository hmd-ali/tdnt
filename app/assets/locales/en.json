{
  "test": "test",
  "form": {
    "errors": {
      "email": {
        "required": "Email is required"
      }
    }
  }
}
