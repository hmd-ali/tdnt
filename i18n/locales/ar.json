{
  "form": {
    "email": "بريد إلكتروني",
    "field_is_required": "{0} مطلوب.",
    "field_must_be_a_string": "يجب أن يكون {0} سلسلة.",
    "password": "كلمة المرور",
    "email_must_be_valid": "يجب أن يكون بريدًا إلكترونيًا صالحًا",
    "password_min_length": "يجب أن تتكون كلمة المرور من 8 أحرف على الأقل"
  },
  "login": "تسجيل الدخول",
  "missing_token": "الرمز مفقود أو غير صالح.",
  "logout": "تسجيل الخروج",
  "welcome_name": "مرحبًا {name}!",
  "enter_username_and_pass": "",
  "login_success": "نجاح تسجيل الدخول!",
  "invalid_username_or_password": "اسم المستخدم أو كلمة المرور غير صالحة",
  "logged_in_route": "تم تسجيل الدخول في الطريق"
}
