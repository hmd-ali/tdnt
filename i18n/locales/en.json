{
  "form": {
    "field_must_be_a_string": "{0} must be a string.",
    "field_is_required": "{0} is required.",
    "email": "Email",
    "password": "Password",
    "email_must_be_valid": "Must be a valid email",
    "password_min_length": "Password must be at least 8 characters long"
  },
  "login": "Login",
  "missing_token": "Token is missing or invalid.",
  "logout": "Logout",
  "enter_username_and_pass": "Enter your username and password",
  "welcome_name": "Welcome, {name}!",
  "invalid_username_or_password": "Invalid username or password",
  "login_success": "Login success!",
  "logged_in_route": "Logged in route",
  "back_home": "Go back home"
}
