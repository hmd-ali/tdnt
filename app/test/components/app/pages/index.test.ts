import { mountSuspended } from "@nuxt/test-utils/runtime"
import { describe, expect, it } from "vitest"

import Index from "~/pages/index.vue"
import { createI18n } from "vue-i18n"
import en from "~~/i18n/locales/en.json"
import ar from "~~/i18n/locales/ar.json"

const i18n = createI18n({
  locale: "en",
  messages: {
    en,
    ar,
  },
})

const mount = () =>
  mountSuspended(Index, {
    global: {
      plugins: [i18n],
    },
  })

describe("pages/index.vue", async () => {
  const locales = useNuxtApp().$i18n.locales

  it("renders the component", async () => {
    const wrapper = await mount()
    expect(wrapper.exists()).toBe(true)
    wrapper.unmount()
  })

  it.each(locales.value.map((l) => [l.code]))(
    "renders in the correct language for the %s locale",
    async (code) => {
      const wrapper = await mount()
      wrapper.vm.$i18n.locale = code
      await wrapper.vm.$nextTick()
      expect(wrapper.html()).toMatchSnapshot()
    },
  )
})
