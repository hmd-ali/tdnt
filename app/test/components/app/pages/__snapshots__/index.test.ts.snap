// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pages/index.vue > renders in the correct language for the ar locale 1`] = `"<div class="flex items-center justify-center py-20 text-2xl font-medium">تم تسجيل الدخول في الطريق</div>"`;

exports[`pages/index.vue > renders in the correct language for the en locale 1`] = `"<div class="flex items-center justify-center py-20 text-2xl font-medium">Logged in route</div>"`;
